common/src/main/java/com/wynntils/features/overlays/InfoBoxFeature.java

```diff
@@ -98,38 +86,6 @@ protected InfoBoxOverlay(
             this.secondsPerRecalculation = secondsPerRecalculation;
         }
 
-        @Override
-        public void render(
-                PoseStack poseStack, MultiBufferSource.BufferSource bufferSource, float partialTicks, Window window) {
-            if (!Models.WorldState.onWorld()) return;
-
-            if (System.nanoTime() - lastUpdate > secondsPerRecalculation * 1e+9) {
-                lastUpdate = System.nanoTime();
-                cachedLines = Managers.Function.doFormatLines(content);
-            }
-
-            float renderX = this.getRenderX();
-            float renderY = this.getRenderY();
-            for (String line : cachedLines) {
-                BufferedFontRenderer.getInstance()
-                        .renderAlignedTextInBox(
-                                poseStack,
-                                bufferSource,
-                                line,
-                                renderX,
-                                renderX + this.getWidth(),
-                                renderY,
-                                renderY + this.getHeight(),
-                                0,
-                                CommonColors.WHITE,
-                                this.getRenderHorizontalAlignment(),
-                                this.getRenderVerticalAlignment(),
-                                this.textShadow);
-
-                renderY += FontRenderer.getInstance().getFont().lineHeight;
-            }
-        }
-
         @Override
         public void renderPreview(
```

Comment:
Maybe the preview can move to TextOverlay as well?

Reply:
Done

Reply:
Hm. I was more thinking like rather having a `getPreviewTemplate()`, and move the `renderPreview` method to the superclass as well. That way the subclasses only need to provide the template and do not really have to bother with the rendering business at all. They just say: "here is my string, go and make it look nice on the screen".

Reply:
(With that said, having the common functionality extracted to `renderTemplate`)

Reply:
Done